cucumber/index.d.ts

```diff
@@ -70,5 +70,139 @@ declare namespace cucumber {
 		Around(code: AroundCode):void;
 		setDefaultTimeout(time:number): void;
 		registerHandler(handlerOption:string, code:(event:any, callback:CallbackStepDefinition) =>void): void;
+		registerListener(listener: EventListener): void;
+	}
+
+	export class EventListener {
+		hear(event: events.Event, callback: ()=>void): void;
+		hasHandlerForEvent(event: events.Event): boolean;
+		buildHandlerNameForEvent(event: events.Event): string;
+		getHandlerForEvent(event: events.Event): EventHook;
+		buildHandlerName(shortName: string): string;
+		setHandlerForEvent(shortName: string, handler: EventListener): void;
+	}
+
+	export function Listener(): EventListener;
+
+	export namespace events {
+
+		interface Event {
+			getName(): string;
+			getPayloadItem(name: string): EventPayload;
+		}
+
+		interface EventPayload {
```

Comment:
Ah yes, that's a good point. A union type might actually work better here, thanks for your suggestion. I'll update the PR tonight.